# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/cookiestore/CookieStoreNotificationWatcherWrapper.h
# Commit: 30d5dfedf252
# Full Hash: 30d5dfedf252f4f10a0d8948e86f4e57f1c4ccf9
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-11 09:21:34
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 8 - CookieStore API - e10s support, r=smaug,cookie-reviewers,timhuang
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D217574
# ==============================================================================

diff -r fc3b4c5ee4b2 -r 30d5dfedf252 dom/cookiestore/CookieStoreNotificationWatcherWrapper.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/cookiestore/CookieStoreNotificationWatcherWrapper.h	Tue Sep 10 16:31:41 2024 +0000
@@ -0,0 +1,39 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=8 sts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef mozilla_dom_CookieStoreNotificationWatcherWrapper_h
+#define mozilla_dom_CookieStoreNotificationWatcherWrapper_h
+
+#include "mozilla/OriginAttributes.h"
+
+namespace mozilla::dom {
+
+class Promise;
+class CookieStore;
+class CookieStoreNotificationWatcher;
+
+class CookieStoreNotificationWatcherWrapper final {
+ public:
+  NS_INLINE_DECL_THREADSAFE_REFCOUNTING(CookieStoreNotificationWatcherWrapper)
+
+  static already_AddRefed<CookieStoreNotificationWatcherWrapper> Create(
+      CookieStore* aCookieStore);
+
+  void ResolvePromiseWhenNotified(const nsID& aOperationID, Promise* aPromise);
+  void ForgetOperationID(const nsID& aOperationID);
+
+ private:
+  CookieStoreNotificationWatcherWrapper() = default;
+  ~CookieStoreNotificationWatcherWrapper();
+
+  void CreateWatcherOnMainThread(bool aPrivateBrowsing);
+
+  RefPtr<CookieStoreNotificationWatcher> mWatcherOnMainThread;
+};
+
+}  // namespace mozilla::dom
+
+#endif /* mozilla_dom_CookieStoreNotificationWatcherWrapper_h */